app/assets/javascripts/application.js

```diff
@@ -375,7 +375,7 @@ $(function() {
 
 function update_puppetclasses(element) {
   var host_id = $("form").data('id')
-  var env_id = $('*[id*=environment_id]').val();
+  var env_id = $('*[name*=environment_id]').val();
```

Comment:
I think I agree that name within a form is OK to search for, I'm not sure that ID is particularly beneficial here.  But perhaps we should be more specific and search for `select` rather than `*`?